apps/examples/connect_test/connect_test_client.c

```diff
@@ -0,0 +1,676 @@
+		return -1;
+	}
+	return 0;
+}
+
+static int broadcast_receive(int sock, char* msg, int msg_len)
+{
+	int nbytes = 0;
+	unsigned int len = sizeof(struct sockaddr_in);
+	int retry_count = 10;
+	struct sockaddr_in sender_addr;
+	while (retry_count--) {
+		nbytes = recvfrom(sock, msg, msg_len, 0, (struct sockaddr*)&sender_addr, &len);
+		if (nbytes > 0) {
+			CT_LOG(TAG, "received message: %s", msg);
+			char *broadcast_mac, *broadcast_ip;
+			broadcast_mac = strtok(msg, " ");
+			broadcast_ip = strtok(NULL, " ");
+			CT_LOG(TAG, "The stored mac of peer is %s", g_recv_mac_buf);
+			if(!strncmp(g_recv_mac_buf, broadcast_mac, strlen(g_recv_mac_buf))) {
+				strncpy(g_tcp_server_address, broadcast_ip, strlen(broadcast_ip)+1);
+				CT_LOG(TAG, "The mac of peer is %s and ip of peer is %s",broadcast_mac, broadcast_ip);
+				CT_LOG(TAG, "server ip %s", g_tcp_server_address);
+				break;
+			} else {
+				continue;
+			}
+		} else {
+			if (errno == EWOULDBLOCK) {
+				CT_LOGE(TAG, "timeout trying again, error %s-%d", strerror(errno), errno);
+				continue;
+			} else {
+				CT_LOGE(TAG, "Receving broadcast message failed");
+				return -1;
+			}
+		}
+	}
+	return 0;
+}
+
+int connect_test_run_client(void)
+{
+	int conn = 0;
+	/*  connect to softAP */
+	CT_LOG(TAG, "connect AP");
+	wifi_manager_ap_config_s ap_config;
+	wm_get_created_softapinfo(&ap_config);
+	wifi_manager_result_e wres = WIFI_MANAGER_SUCCESS;
+	int retry_connect = MAX_AP_CON_RETRY;
+	while(retry_connect) {
+		wres = wifi_manager_connect_ap(&ap_config);
+		if (wres != WIFI_MANAGER_SUCCESS) {
+			CT_LOGE(TAG, "connect AP fail %d", wres);
+			CT_LOG(TAG, "add 100 ms sleep, gives time to driver to stabilise");
+			usleep(100000);
+		} else {
+			break;
+		}
+		retry_connect--;
+	}
+	
+	if (wres != WIFI_MANAGER_SUCCESS) {
+		return -1;
+	}
+
+	/*  wait connect event */
+	CT_LOG(TAG, "wait connect success event");
+	WO_TEST_WAIT(conn, g_wo_queue);
+	
+	CT_LOG(TAG, "add 2 seconds sleep, let peer machine go in receive mode");
+	sleep(1);
+	char mac_address[6];
+ 	if (netlib_getmacaddr("wlan0", mac_address) != 0) {
+		CT_LOGE(TAG, "MAC Get Error");
+                return 0;
+        }
+        snprintf((char *)g_mac_buf, 13, "%02X%02X%02X%02X%02X%02X",
+               mac_address[0], mac_address[1], mac_address[2], 
+               mac_address[3], mac_address[4], mac_address[5]);
+        CT_LOG(TAG, "Device ID: %s", g_mac_buf);
+	
+	/* Data transfer using UDP or DTLS */
+	if(g_WM_IS_DTLS) {
+		wm_test_dtls_client(WT_UDP_SERVER_ADDR, g_WM_DTLS_DATA_SIZE, g_mac_buf, g_recv_mac_buf, MAC_BUF_SIZE);
+	} else {
+		_udp_client(g_WM_DTLS_DATA_SIZE);
+	}
+
+	CT_LOG(TAG, "wait for turnoff softap");
+	WO_TEST_WAIT(conn, g_wo_queue);
+	
+	/*  connect to AP */
+	CT_LOG(TAG, "connect AP");
+	wifi_manager_ap_config_s apconfig;
+	wm_get_apinfo(&apconfig, g_WM_AP_SSID, g_WM_AP_PASSWORD, g_WM_AP_AUTH, g_WM_AP_CRYPTO);
+	retry_connect = MAX_AP_CON_RETRY;
```

Comment:
It's repeated code. how about to make it function.

Reply:
Changes Done as suggested